### ISSUE TITLE
Add design principles to Storybook

### ISSUE THREAD

**Issue by rutvigupta-design** [department-of-veterans-affairs/caseflow#16018]
# Caseflow Design Principles
### 💪🏽 Prioritize accessibility early on
The VA is the largest employer of people with disabilities and considering accessibility cannot be an afterthought for our team. We make space for accessibility from the very moment we kick a project off, and we advocate for accessible practices and design throughout the lifecycle of the project. We commit to designing for present and future users with disabilities.  

#### What this looks like in practice:
- When planning the research and design for your project, we account for accessibility. Our research and design methods account for accommodations. 
- We design for keyboard-interactions first and build in time and space to consider accessibility in the creation of our design tickets and planning.
- We advocate for continuous accessibility testing and growing our Caseflow team's skills during rituals such as dogfooding and UAT. 
- We seek out and share resources that can help us grow our accessibility expertise. 
- We refer to an accessibility checklist continuously: 
  - Are our contrast ratios high enough for users who are colorblind?
  - Is our language plain and easy to understand?
  - Can users navigate our site using only the keyboard?
  - Can users use a screen reader to access the page content?
  - Can users quickly understand the main points of our content?
  - Can users easily interpret content associated with graphic elements?
  - Can users easily understand and complete key tasks?
  - Are we testing your service with a broad range of users?
  - Do we know your agency accessibility team?
  - Is our site organized such that everyone can navigate it easily?
  - Are we using accessibility testing tools?
  - Did our accessibility testing tools provide accurate results?

### 🤝 Use participatory means
Our users shape Caseflow through collaboration and co-creation. We value, respect, and center our users' expertise and capabilities and strive to create safe spaces for our users to fully participate in. Power dynamics (including our own) are acknowledged and we strive to build trust with our collaborators. We speak plainly and transparently, avoiding jargon. We take the time to explain our goals and methods plainly and make it clear how their contributions will be used moving forward. We also continuously and sustainably share our progress and results with our users. 

#### What this looks like in practice:
- We continuously engage with a varied and diverse set of real users and strive to build strong and mutually beneficial relationships with them. 
- We seek out and/or practice methods and formats that can encourage collaboration and co-creation, when possible. 
- We design our research and design sessions with our users' needs in mind.
- We take the time to explain our goals and methods plainly. 
- We reach back out to users to share progress and results through the lifecycle of our projects.  

### 🏅Radically invest in getting it right
Trust is earned, and we strive to ensure we are designing our interactions, both within the product, and with our government partners with intention. We move at the speed of trust and are constantly careful to not compromise or harm the trust we have built up by simply prioritizing speed. Whether it is a meeting with our government partners or an interaction design decision, we pay close attention to all the details – small or big. 

#### What this looks like in practice: 
- We enact our design principles with each interaction to meet our user's expectations of functionality, behavior, consistency, error prevention and recovery, and accuracy and reliability of data.
- We take the time to consider unintended consequences.
- We design all our interactions with our government partners carefully, and approach these interactions with professionalism and thoughtfulness. 
- We present our designs and decisions at various stages, to internal and external audiences to ensure we are on the right track and are covering our blind spots. 

### 🔎 Seek out opportunities to deliver incremental value
Caseflow is continuously maturing, and we must strive to deliver with focus. At times of prioritization, we are intentional about how we are delivering value to our end users. As designers, we constantly seek out ways to improve our user experiences – we are our users' advocates. 

#### What this looks like in practice: 
- When thinking about business processes, we look to our research to identify opportunities to streamline or enhance experiences.
- We surface tangential needs or pain points voiced by our users to the appropriate people. 
- We think about flows and user experiences holistically and identify moments or touchpoints that can be improved.   

### 💬 Overshare and uncloud
The appeals process can often be opaque to VA employees, Veterans, and representatives alike - both in terms of overarching legal requirements and transparency into the bureaucratic minutiae of who acted upon an appeal when.

To guide mastery of the appeals process and of Caseflow (which is one piece of the larger appeals process), we continuously promote clarity and transparency and avoid ambiguity. We guide users through their options and next steps in Caseflow, and we clearly explain the legal reasons behind those options and steps. We capture information on who performs actions in Caseflow and when, in order to promote transparency and ownership.

#### What this looks like in practice: 
- We make the next steps required by Caseflow and/or the law to move an appeal through the process (1) apparent and (2) clearly explained.
- We transparently record the details of actions that affect an appeal’s movement through the process in Caseflow. 
- We give specific and actionable messages – we do not default to vague or default messaging. 
- We consider users who are new to the Board and make them feel like geniuses. 

### 🔃 Prioritize flexibility 
Since Caseflow is an enterprise application that services a wider range of users, we ensure Caseflow is flexible and forgiving. A single appeal may touch several Board users in Caseflow, and appeals should never get stuck or blocked (or worse – lost), even when that means going back a step or two. We must prevent errors before they occur through clear, informative guidance and design constraints. However, when mistakes are inevitably made, we offer users flexible and forgiving pathways to recover. Board business processes also evolve, and Caseflow responds to the Board's evolving needs. 

#### What this looks like in practice: 
- When designing, we give users ways to edit or undo actions or edit data they’ve added to the system.
- We match the level of friction at decision points with the level of risk and irrevocability. 
- We consider striking the right balance between alerting and informing users of trade-offs and implications in order to avoid alert fatigue. 
- We make Caseflow feel quick and responsive when possible.  

### ⚖️ Consistency over invention
Default to re-using established patterns. Caseflow should feel consistent and familiar across our varied products. While some products like Caseflow Reader will have specialized components to serve specific user needs, we don't reinvent the wheel. As much as possible, we re-use patterns — from layouts and tables to form elements and navigation — throughout Caseflow to create a consistent experience.

#### What this looks like in practice: 
- When designing, we consider what established visual and functional patterns are serving a similar user need. 
- When we want to introduce a new pattern, we consider how we can design it to maximize re-use across Caseflow in the future.

Use language concisely and consistently. Copy content is a large part of Caseflow, and our voice, tone, and language should feel consistent and familiar across our different products. We say things simply, professionally, and are clear and actionable. We use terms and define them consistently across products.  

#### What this looks like in practice: 
- When designing, we consider what language other parts of the product are using. 
- We use language and terms that are familiar to our users. 
- When possible, we personalize content or messages to reference the actual Veteran or appellant.
- As a design team, we standardize terms and language so we are consistent.  
- As a design team, we use spaces such as our team-wide design sync, design critique etc. to align on our use of language.

**Comment by rutvigupta-design** [comment]
@nicholer does this look okay? Feel free to go in and make any changes!

**Comment by nicholer** [comment]
@rutvigupta-design looks good to me 🎉

I just made small updates to the heading hierarchy, so Caseflow Design Principles is the h1

**Comment by rutvigupta-design** [comment]
Perfect I'll tag JC!
